nomad/structs/funcs_test.go

```diff
@@ -716,6 +716,82 @@ func TestScoreFitBinPack(t *testing.T) {
 	}
 }
 
+func TestAllocsFit_MaxNodeAllocs(t *testing.T) {
+	ci.Parallel(t)
+	baseAlloc := &Allocation{
+		AllocatedResources: &AllocatedResources{
+			Tasks: map[string]*AllocatedTaskResources{
+				"web": {
+					Cpu: AllocatedCpuResources{
+						CpuShares:     1000,
+						ReservedCores: []uint16{},
+					},
+					Memory: AllocatedMemoryResources{
+						MemoryMB: 1024,
+					},
+				},
+			},
+			Shared: AllocatedSharedResources{
+				DiskMB: 5000,
+				Networks: Networks{
+					{
+						Mode:          "host",
+						IP:            "10.0.0.1",
+						ReservedPorts: []Port{{Label: "main", Value: 8000}},
+					},
+				},
+				Ports: AllocatedPorts{
+					{
+						Label:  "main",
+						Value:  8000,
+						HostIP: "10.0.0.1",
+					},
+				},
+			},
+		},
+	}
+
+	testCases := []struct {
+		name        string
+		allocations []*Allocation
+		expectErr   bool
+		maxAllocs   int
+	}{
+		{
+			name:        "happy_path",
+			allocations: []*Allocation{baseAlloc},
+			expectErr:   false,
+			maxAllocs:   2,
+		},
+		{
+			name:        "too many allocs",
+			allocations: []*Allocation{baseAlloc, baseAlloc, baseAlloc},
+			expectErr:   true,
+			maxAllocs:   2,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			n := node2k()
+			n.NodeMaxAllocs = &NodeMaxAllocs{tc.maxAllocs}
+			fit, dim, used, err := AllocsFit(n, tc.allocations, nil, false)
+			if !tc.expectErr {
+				must.NoError(t, err)
+				must.True(t, fit)
+				must.Eq(t, 1000, used.Flattened.Cpu.CpuShares)
+				must.Eq(t, 1024, used.Flattened.Memory.MemoryMB)
+			} else {
+				must.Error(t, err)
+				must.False(t, fit)
+				must.StrContains(t, dim, "max allocation exceeded")
+				must.StrContains(t, err.Error(), "plan exceeds max allocation")
```

Comment:
`ErrorContains` contains a nil-check for the error, so we can combine these:

```suggestion
				must.ErrorContains(t, err, "plan exceeded max allocation")
				must.False(t, fit)
				must.StrContains(t, dim, "max allocation exceeded")
```